### PR TITLE
Added support for enabling NTP service on PI master and syncing time on agents

### PR DESCRIPTION
**Pull request by greglarkin**

### CODE REVIEW

File: manifests/course/puppetize.pp [hunk 1]
```diff
@@ -14,6 +14,10 @@
       mode  => '0644',
     }
 
+    class { 'classroom':
```
Comment (binford2k) [inline manifests/course/puppetize.pp]: This pulls in a *ton* of dependent classes. I'd rather not if we can avoid it. Did you try just `classroom::virtual`?

Comment (greglarkin) [inline manifests/course/puppetize.pp]: I tried adding some code to classroom::virtual to manage time servers, but it caused some errors. Then I realized I'd need gitea as well. I'll go back and look at it again and see if I can figure out a different solution.

**Comment by greglarkin** [comment]
Oh yeah, now I see it. I included `classroom::master::time`, but it needs access to `$classroom::offline`, because it assumes that it is included by class `classroom`. Rather than figure out how to break that linkage, I opted for the easy way out. I can look at making the code more general, if you like.

**Comment by binford2k** [comment]
@greglarkin well, let's take a step back and see what we're trying to accomplish in the first place. We explicitly left time sync out of virtual courses since it didn't seem to matter. All nodes are running on the same machine so they're using the same hardware clock.

**Comment by greglarkin** [comment]
Ok, then maybe the correct fix is to just modify setup_classroom.sh to avoid time sync. I was just pushing forward trying to fix the script bombing out without considering modifying the script itself.

**Comment by binford2k** [comment]
What will the process of provisioning a node be like? Do we even need to run `setup_classroom.sh`?

I feel like @samuelson had talked about a Puppetfactory plugin for it. That'd be something like https://github.com/puppetlabs/puppetfactory/blob/master/puppetfactory/lib/puppetfactory/plugins/shell_user.rb

**Comment by greglarkin** [comment]
At the moment, a rake task on the CIAB machine provisions all of the student VMs with Vagrant. Those aren't hooked into Puppetfactory yet, but perhaps I need to look at how to do that. Since that's a heavier process than provisioning a Docker container, I assume we don't want to do that with the "New User" button but just have them all ready to go? I do need to figure out how to assign the different Vagrant boxes to the students.

**Pull request closed without merging**